InvalidLocationConstraint: The specified location-constraint is not valid

Trying to run the benchmark tool from an EC2 instance on an existing bucket:
```
[ec2-user@ip-172-31-2-167 ~]$ ./s3-benchmark    -bucket-name dhavivresearch -full -region us-east-1

--- SETUP --------------------------------------------------------------------------------------------------------------------

panic: Filed to create S3 bucket: InvalidLocationConstraint: The specified location-constraint is not valid
        status code: 400, request id: 41997BE73BC03A06, host id: ZNEaOvY5TyFs0PR3xh73aUAPVg6y18y9iV/bzJH4TJYiQ8/sKPk+wIqReWkRtpH3/wV+oV53Sns=

goroutine 1 [running]:
main.setup()
        /Users/daniel/go/src/github.com/dvassallo/s3-benchmark/main.go:229 +0xa35
main.main()
        /Users/daniel/go/src/github.com/dvassallo/s3-benchmark/main.go:113 +0x44
```

--- a/README.md
+++ b/README.md
@@ -1,4 +1,3 @@
-
 # S3 Benchmark
 
 Your [Amazon S3](https://aws.amazon.com/s3/) performance depends on 3 things:

--- a/README.md
+++ b/README.md
@@ -1,3 +1,4 @@
+
 # S3 Benchmark
 
 Your [Amazon S3](https://aws.amazon.com/s3/) performance depends on 3 things:
@@ -53,6 +54,27 @@ Or run the full test (takes a few hours):
 ```
 
 See [this](https://github.com/dvassallo/s3-benchmark/blob/master/main.go#L123-L134) for all the other options.
+
+### Build
+
+1. Install [Go](https://golang.org/)
+    ```
+    sudo apt-get install golang-go
+    ```
+    or
+    ```
+    sudo yum install go
+    ```
+    may work too. 
+    
+2. Setup Go environment variables (Usually GOPATH and GOBIN) and test Go installation 
+3. Clone the repo
+4. Install [```dep```](https://golang.github.io/dep/) 
+	```
+	go get -u github.com/golang/dep/cmd/dep
+	```
+5. Go to source directory and run ```dep ensure```
+6. Run ```go run main.go```
 
 ## S3 to EC2 Bandwidth
 

--- a/main.go
+++ b/main.go
@@ -218,7 +218,7 @@ func setup() {
 	createBucketReq := s3Client.CreateBucketRequest(&s3.CreateBucketInput{
 		Bucket: aws.String(bucketName),
 		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
-			LocationConstraint: s3.BucketLocationConstraint(region),
+			LocationConstraint: s3.NormalizeBucketLocation(s3.BucketLocationConstraint(region)),
 		},
 	})
 	

--- a/main.go
+++ b/main.go
@@ -221,6 +221,14 @@ func setup() {
 			LocationConstraint: s3.BucketLocationConstraint(region),
 		},
 	})
+	
+	// AWS S3 has this peculiar issue in which if you want to create bucket in us-east-1 region, you should NOT specify 
+	// any location constraint. https://github.com/boto/boto3/issues/125
+	if strings.ToLower(region) == "us-east-1" {
+		createBucketReq = s3Client.CreateBucketRequest(&s3.CreateBucketInput{
+			Bucket: aws.String(bucketName),
+		})
+	}
 
 	_, err := createBucketReq.Send()
 

